test(openchallenges): improve organization service model ImageResponse and OrganizationMapper unit test coverage

mdsage1 opened the pull request:
## Description

Changes include packages w/a single class.
Sonar unit test coverage has been increased to 12% for model/mapper package. This package has a single class, OrganizationMapper.
Sonar unit test coverage has been increased to 100% for service/model/rest/response package. This package has a single class, ImageResponse.
The OrganizationServiceApplication class in the organization/service package is the only class in that package but the only uncovered code would be categorized as an integration test while the focus here is on unit testing.

A comment was also added to OrganizationServiceApplication for the existing integration test. This confused me and I had to deep dive into what was happening to find out that it was an integration test rather than a unit test.

## Related Issue

#2063 
#1528 
#2348 

## Changelog

See `files Changed` section

These changes have not increased the Sonar coverage above the 15%,9.5% according to the [Code Quality Dashboard](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/CODE_QUALITY.md), achieved with the exception unit test PR #2348 but, different components, methods etc., of these packages' classes are being tested by the unit tests added. These tests were written last week and debugged this week. The PR has been released after the approval of the exception unit test PR on 2023-12-04.

## Preview

The Sonar coverage increase for model/mapper:
![image](https://github.com/Sage-Bionetworks/sage-monorepo/assets/122999770/d5ff8607-3144-479d-8193-4148d17215d5)

The Sonar coverage increase for service/model/rest/response:
![image](https://github.com/Sage-Bionetworks/sage-monorepo/assets/122999770/8aa2ce63-7777-46f1-98a2-f7ea41756e0f)

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/mapper/OrganizationMapperTest.java:
```diff
@@ -0,0 +1,111 @@
+  public void ConvertToEntity_ShouldReturnDtoProperties_WhenDtoPropertiesPassed() {
+
+    // create new Dto with properties to be copied to organization entity
+    OrganizationDto user = new OrganizationDto();
```
Rename variable

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/mapper/OrganizationMapperTest.java:
```diff
@@ -0,0 +1,111 @@
+
+    OrganizationMapper mapper = new OrganizationMapper();
+
+    OrganizationEntity organizationEntity = mapper.convertToEntity(user);
```
You could simplify by renaming `organizationEntity` to `entity` as there is only one entity defined here. Also to be consistent with the naming you use for the mapper. Consider other places where this comment could apply.

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/OrganizationServiceApplicationTests.java:
```diff
@@ -6,6 +6,7 @@
 @SpringBootTest
 class OrganizationServiceApplicationTests {
 
+  // integration test
```
Are you sure that this is an integration test? Or this is also used to prepare the context for the unit tests?

mdsage1 replied:
I'm not sure. That's the conclusion that I came to after looking into it. If it's used to prepare the context for the unit tests I will will change the comment.

tschaffter reviewed (changes requested):
Looks good overall! See comments below

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/OrganizationServiceApplicationTests.java:
```diff
@@ -6,6 +6,7 @@
 @SpringBootTest
 class OrganizationServiceApplicationTests {
 
+  // prepare the context for the unit tests
```
Don't take my words for it. If unsure about what this line does, remove this comment.

mdsage1 replied:
Ok I'll just remove the comment.

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/mapper/OrganizationMapperTest.java:
```diff
@@ -0,0 +1,111 @@
+    String retrievedAcronym = entity.getAcronym();
+
+    Long retrievedId2 = organizationDto.getId();
+    String retrievedName2 = organizationDto.getName();
+    String retrievedEmail2 = organizationDto.getEmail();
+    String retrievedLogin2 = organizationDto.getLogin();
+    String retrievedAvatarKey2 = organizationDto.getAvatarKey();
+    String retrievedWebsiteUrl2 = organizationDto.getWebsiteUrl();
+    Integer retrievedChallengeCount2 = organizationDto.getChallengeCount();
+    String retrievedDescription2 = organizationDto.getDescription();
+    String retrievedAcronym2 = organizationDto.getAcronym();
```
See my comment about "actual" Vs. "expected" in the other open PR. Here you would rename "retrieved" by "actual".

mdsage1 replied:
Got it thanks! Updating it now.

mdsage1 commented:
1. Removed the comment regarding the integration test in [service/OrganizationServiceApplicationTests](apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/OrganizationServiceApplicationTests.java)
2. Renamed organizationEntity to entity in [model/mapper/OrganizationMapperTest.java](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2388/files/0ae9260076f538b323a32b28239e420424dc7c4f#diff-a73f199d2fc9e997cc9994799e31c3c7c3e93aabcb1e78764ef3bbc2039132cd)
3. Renamed user variable to organizationDto in [service/model/mapper/OrganizationMapperTest.java](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2388/files/0ae9260076f538b323a32b28239e420424dc7c4f#diff-a73f199d2fc9e997cc9994799e31c3c7c3e93aabcb1e78764ef3bbc2039132cd)
4. Changed the variable names that used `retrieved` to `actual` throughout [model/mapper/OrganizationMapperTest.java](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2388/files/0ae9260076f538b323a32b28239e420424dc7c4f#diff-a73f199d2fc9e997cc9994799e31c3c7c3e93aabcb1e78764ef3bbc2039132cd)

tschaffter commented:
> 1. Removed the comment regarding the integration test in [service/OrganizationServiceApplicationTests](https://github.com/Sage-Bionetworks/sage-monorepo/pull/apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/OrganizationServiceApplicationTests.java)
> ...

Let's keep the discussion in their respective thread. When you address a comment from a reviewer, you can either close it or comment + close it if you have addressed it. If work is in progress or clarification is needed, keep the comment thread open. Thanks!

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/mapper/OrganizationMapperTest.java:
```diff
@@ -0,0 +1,111 @@
+  public void ConvertToEntity_ShouldReturnDtoProperties_WhenDtoPropertiesPassed() {
+
+    // create new Dto with properties to be copied to organization entity
+    OrganizationDto dTo = new OrganizationDto();
```
Don't make up new spellings: here use `dto` or `organization`.

mdsage1 replied:
Ok I thought I needed to capitalize the T in to since it was a variable name.

tschaffter replied:
The only difference between a variable name and a class name is the first letter (capitalized for class name).

mdsage1 replied:
Updated the capitalization of `dTo` to all lowercase.

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/rest/response/ImageResponseTest.java:
```diff
@@ -0,0 +1,21 @@
+    String retrievedUrl = imageResponse.getUrl();
+
+    // Confirm that the same image url was retrieved that was set
+    Assertions.assertEquals(imageUrl, retrievedUrl);
```
Another "retrieved" to replace.

mdsage1 replied:
Replaced `retrieved` with `actual`.

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/mapper/OrganizationMapperTest.java:
```diff
@@ -0,0 +1,111 @@
+
+    OrganizationEntity entity = mapper.convertToEntity(dto);
+
+    Long actualId = entity.getId();
+    String actualName = entity.getName();
+    String actualEmail = entity.getEmail();
+    String actualLogin = entity.getLogin();
+    String actualAvatarKey = entity.getAvatarKey();
+    String actualWebsiteUrl = entity.getWebsiteUrl();
+    Integer actualChallengeCount = entity.getChallengeCount();
+    String actualDescription = entity.getDescription();
+    String actualAcronym = entity.getAcronym();
+
+    Long actualId2 = dto.getId();
+    String actualName2 = dto.getName();
+    String actualEmail2 = dto.getEmail();
+    String actualLogin2 = dto.getLogin();
+    String actualAvatarKey2 = dto.getAvatarKey();
+    String actualWebsiteUrl2 = dto.getWebsiteUrl();
+    Integer actualChallengeCount2 = dto.getChallengeCount();
+    String actualDescription2 = dto.getDescription();
+    String actualAcronym2 = dto.getAcronym();
+
+    // verify the entity properties were copied
+    Assertions.assertEquals(actualId, actualId2);
+    Assertions.assertEquals(actualName, actualName2);
+    Assertions.assertEquals(actualEmail, actualEmail2);
+    Assertions.assertEquals(actualLogin, actualLogin2);
+    Assertions.assertEquals(actualAvatarKey, actualAvatarKey2);
+    Assertions.assertEquals(actualWebsiteUrl, actualWebsiteUrl2);
```
The code would be more compact and readable if you do:

```
Assertions.assertEquals(dto,getId(), entity.getId());
...
```

mdsage1 replied:
Updated the code to reflect the requested changes.
